### PR TITLE
fix: Loan Repayment entry cancellation on salary slip cancel

### LINKED ISSUE #24543: Canceling and deleting a draft salary slip should reset loan repayments-v13-beta
To replicate:

- Create an employee loan, disburse, then run loan interest accrual.
- Create a salary slip via payroll entry, or manually. The salary slip will pick the loan.
- Now, delete the salary slip.
- On creating the salary slip again, it won't pick the load repayment.

### PR DESCRIPTION
**Pull request by deepeshgarg007**
Closes: https://github.com/frappe/erpnext/issues/24543

### CODE REVIEW

File: erpnext/loan_management/doctype/loan_repayment/loan_repayment.py [hunk 1]
```diff
@@ -265,6 +274,10 @@ def make_gl_entries(self, cancel=0, adv_adj=0):
 		if gle_map:
 			make_gl_entries(gle_map, cancel=cancel, adv_adj=adv_adj, merge_entries=False)
 
+	# def repost_future_loan_interest_accruals(self):
+	# 	future_lias = frappe.db.get_all("Loan Interest Accrual", {"docstatus": 1, "posting_date": (">", self.posting_date)})
+	# 	if future_lias:
+
```
Comment (sagarvora) [inline erpnext/loan_management/doctype/loan_repayment/loan_repayment.py]: Please remove commented code.

File: erpnext/payroll/doctype/salary_slip/salary_slip.py [hunk 2]
```diff
@@ -1050,7 +1050,7 @@ def make_loan_repayment_entry(self):
 			repayment_entry.save()
 			repayment_entry.submit()
 
-			loan.loan_repayment_entry = repayment_entry.name
+			frappe.db.set_value("Salary Slip Loan", loan.name, "loan_repayment_entry", repayment_entry.name)
```
Comment (sagarvora) [inline erpnext/payroll/doctype/salary_slip/salary_slip.py:1056]: Why is `db.set_value` being preferred here? Doesn't the existing code work just fine? Why have multiple DB calls?

File: erpnext/loan_management/doctype/loan_repayment/loan_repayment.py [hunk 3]
```diff
@@ -30,6 +31,7 @@ def before_submit(self):
 	def on_submit(self):
 		self.update_paid_amount()
 		self.make_gl_entries()
+		#self.repost_future_loan_interest_accruals()
```
Comment (sagarvora) [inline erpnext/loan_management/doctype/loan_repayment/loan_repayment.py]: Commented code.

File: erpnext/loan_management/doctype/loan_repayment/loan_repayment.py [hunk 4]
```diff
@@ -63,6 +65,13 @@ def set_missing_values(self, amounts):
 		if amounts.get('due_date'):
 			self.due_date = amounts.get('due_date')
 
+	def check_future_entries(self):
+		future_repayment_date = frappe.db.get_value("Loan Repayment", {"posting_date": (">", self.posting_date),
+			"docstatus": 1}, 'posting_date')
+
+		if future_repayment_date:
```
Comment (sagarvora) [inline erpnext/loan_management/doctype/loan_repayment/loan_repayment.py]: (Minor) You can use `frappe.db.exists` here. Also, no need to request `posting_date` specifically. The default `name` can work too.

**Review by sagarvora — DISMISSED** [review]

File: erpnext/loan_management/doctype/loan_repayment/loan_repayment.py (hunk 1 above)
Comment (deepeshgarg007) [reply erpnext/loan_management/doctype/loan_repayment/loan_repayment.py]: I just added this to check for test cases, this PR is not complete yet, will uncomment and complete those functions. Added a dont-merge lable

File: erpnext/payroll/doctype/salary_slip/salary_slip.py (hunk 2 above)
Comment (deepeshgarg007) [reply erpnext/payroll/doctype/salary_slip/salary_slip.py:1056]: No it wasn't, this function is called on submit, the doc is already validated and saved hence the field is not updated using the current code.

File: erpnext/loan_management/doctype/loan_repayment/loan_repayment.py (hunk 4 above)
Comment (deepeshgarg007) [reply erpnext/loan_management/doctype/loan_repayment/loan_repayment.py]: I am using the posting_date in the below error message, hence queried posting_date

Comment (sagarvora) [reply erpnext/loan_management/doctype/loan_repayment/loan_repayment.py]: Sorry, missed that

File: erpnext/payroll/doctype/salary_slip/salary_slip.py (hunk 2 above)
Comment (sagarvora) [reply erpnext/payroll/doctype/salary_slip/salary_slip.py:1053]: Hmm, maybe you can add a `before_submit` method instead.

**Pull request merged**